### DISCUSSION TITLE
O que estão utilizando de mais morderno para eCommerces?

### DISCUSSION THREAD

**Discussion by matheusbronca** [frontendbr/forum#1918] [Discussão]
Peguei um e-Commerce pra fazer e fui de Woocommerce, mas meu Deus do céu parece que abri as portas do inferno (achei muito desorganizado), outra coisa como o back é em PHP eu não achei tão performático assim. O que vocês tem usado pra entregar e-commerces performáticos e escaláveis? (E que seja amigável para que pessoas que não entendem de tecnologia possam gerenciá-los).

Quem puder colaborar eu agradeço o/

**Comment by ninetails** [comment]
Opa, ouvi que tem uma versão do [Next.js para ecommerce](https://nextjs.org/commerce) mas não cheguei a testar ou baixar para ver. E também já ouvi falar de uns poucos sites na Europa usando [Gatsby](https://www.gatsbyjs.com/tutorial/ecommerce-tutorial/).

Mas aí creio que é só o front-end edge, não sei se ajuda em algo ou se ainda requer um back-end integrando. E frameworks de ecommerce para back-end headless ou SAAS estou bem por fora pra falar a verdade, não vou conseguir ajudar neste quesito.

**Comment by matheusbronca** [comment]
> Opa, ouvi que tem uma versão do [Next.js para ecommerce](https://nextjs.org/commerce) mas não cheguei a testar ou baixar para ver. E também já ouvi falar de uns poucos sites na Europa usando [Gatsby](https://www.gatsbyjs.com/tutorial/ecommerce-tutorial/).
> 
> Mas aí creio que é só o front-end edge, não sei se ajuda em algo ou se ainda requer um back-end integrando. E frameworks de ecommerce para back-end headless ou SAAS estou bem por fora pra falar a verdade, não vou conseguir ajudar neste quesito.

Eu também cheguei a dar uma olhada no Next.js ecommerce mas parece ser um projeto embrionário, ainda está longe de estar 100%, mas estou ansioso para experimentar quando estiver em uma release mais completa. Obrigado pela contribuição o/

**Comment by kivervinicius** [comment]
@matheusbronca Hoje ainda compensa desenvolver um? Soluções de terceiros não seriam mais interessantes? Estilo loja integrada ou tray.

**Comment by portothree** [comment]
Next.js com Saleor esta me surpreendendo positivamente, recomendo.

**Comment by matheusbronca** [comment]
> @matheusbronca Hoje ainda compensa desenvolver um? Soluções de terceiros não seriam mais interessantes? Estilo loja integrada ou tray.

Eu acredito que são opções diferentes, dá pra fugir de alguns custos e ter um pouco mais de liberdade na construção do e-commerce. Porém eu sou novato nesse negócio de e-commerce, então talvez eu esteja falando asneira, se houver pessoas com mais experiência que eu por favor colabore com uma resposta.

**Comment by matheusbronca** [comment]
> Next.js com Saleor esta me surpreendendo positivamente, recomendo.

Não conhecia, vou dar uma olhadinha, obrigado por colaborar

**Comment by karluslima** [comment]
> > @matheusbronca Hoje ainda compensa desenvolver um? Soluções de terceiros não seriam mais interessantes? Estilo loja integrada ou tray.
> 
> Eu acredito que são opções diferentes, dá pra fugir de alguns custos e ter um pouco mais de liberdade na construção do e-commerce. Porém eu sou novato nesse negócio de e-commerce, então talvez eu esteja falando asneira, se houver pessoas com mais experiência que eu por favor colabore com uma resposta.

Soluções de terceiros, geralmente são mais robustas e mais estáveis. Tudo depende do que o cliente precisa no e-commerce dele. Essa solução em next por exemplo, acredito que não atenderia as necessidades de players como Boticário, Renner, Decatlhon, Xiaomi, sem ter um desenvolvimento de longos meses e testes. 
Plataforma de e-commerce costumam ser fechadas quando se fala de back-end, então muitas vezes é mais complicado criar novas funcionalidades.

**Comment by ninetails** [comment]
> > > @matheusbronca Hoje ainda compensa desenvolver um? Soluções de terceiros não seriam mais interessantes? Estilo loja integrada ou tray.
> > 
> > 
> > Eu acredito que são opções diferentes, dá pra fugir de alguns custos e ter um pouco mais de liberdade na construção do e-commerce. Porém eu sou novato nesse negócio de e-commerce, então talvez eu esteja falando asneira, se houver pessoas com mais experiência que eu por favor colabore com uma resposta.
> 
> Soluções de terceiros, geralmente são mais robustas e mais estáveis. Tudo depende do que o cliente precisa no e-commerce dele. Essa solução em next por exemplo, acredito que não atenderia as necessidades de players como Boticário, Renner, Decatlhon, Xiaomi, sem ter um desenvolvimento de longos meses e testes.
> Plataforma de e-commerce costumam ser fechadas quando se fala de back-end, então muitas vezes é mais complicado criar novas funcionalidades.

Grandes players ou varejistas se não optam por ter o próprio ecommerce acabam se associando a um marketplace (ou dependendo, ambos). E geralmente possuem fundos pra poder investir num próprio.

Creio que de um tempo pra cá pararam de fazer frameworks voltados a ecommerce, muitos realmente se apoiando em soluções SASS. Faz tempo que não ouço falar nada de um "novo Magento", framework open que faz tudo (porém com core melhor programado 😆 )

**Comment by matheusbronca** [comment]
> > > > @matheusbronca Hoje ainda compensa desenvolver um? Soluções de terceiros não seriam mais interessantes? Estilo loja integrada ou tray.
> > > 
> > > 
> > > Eu acredito que são opções diferentes, dá pra fugir de alguns custos e ter um pouco mais de liberdade na construção do e-commerce. Porém eu sou novato nesse negócio de e-commerce, então talvez eu esteja falando asneira, se houver pessoas com mais experiência que eu por favor colabore com uma resposta.
> > 
> > 
> > Soluções de terceiros, geralmente são mais robustas e mais estáveis. Tudo depende do que o cliente precisa no e-commerce dele. Essa solução em next por exemplo, acredito que não atenderia as necessidades de players como Boticário, Renner, Decatlhon, Xiaomi, sem ter um desenvolvimento de longos meses e testes.
> > Plataforma de e-commerce costumam ser fechadas quando se fala de back-end, então muitas vezes é mais complicado criar novas funcionalidades.
> 
> Grandes players ou varejistas se não optam por ter o próprio ecommerce acabam se associando a um marketplace (ou dependendo, ambos). E geralmente possuem fundos pra poder investir num próprio.
> 
> Creio que de um tempo pra cá pararam de fazer frameworks voltados a ecommerce, muitos realmente se apoiando em soluções SAAS. Faz tempo que não ouço falar nada de um "novo Magento", framework open que faz tudo (porém com core melhor programado 😆 )

Você sabe me informar quais as soluções SAAS mais populares aqui no Brasil?

**Comment by ninetails** [comment]
> Você sabe me informar quais as soluções SASS mais populares aqui no Brasil?

Infelizmente não vou conseguir ajudar 😞 malz man....

Trampo em ecommerce faz um tempo mas sempre fazendo _inhouse_. E como n faço freela, não manjo do que a galera anda adotando por aqui n :\

acho que a galera aí nos outros comments deve manjar mais rs ou espera algum profile fake vir fazer propaganda uhaeuheau

**Comment by kivervinicius** [comment]
Então comentei por que tenho para fazer um, porém são tantas coisas que acho que não vale a "hora/programa" vs "cliente comum" um SAAS seria perfeito, to indo quase para Tray pois la eu vi que tem como customizar várias coisas, mas quando sair algo do comum lascou.

**Comment by ninetails** [comment]
opa, falei errado auheuaheuhae mania de front, é <abbr title="Software-As-A-Service">SAAS</abbr> mas acho que todo mundo entendeu rs

Englobem tb os outros, tipo Plataform As A Service quando falo sobre SAAS

**Comment by matheusbronca** [comment]
> Então comentei por que tenho para fazer um, porém são tantas coisas que acho que não vale a "hora/programa" vs "cliente comum" um SASS seria perfeito, to indo quase para Tray pois la eu vi que tem como customizar várias coisas, mas quando sair algo do comum lascou.

To sentindo na pele isso que você falou, qualquer coisa que saia do comum é um pesadelo pra implementar, e ai acaba não valendo a pena ($) desenvolver pro público "normal".

**Comment by lucas-barbosa** [comment]
> To sentindo na pele isso que você falou, qualquer coisa que saia do comum é um pesadelo pra implementar, e ai acaba não valendo a pena ($) desenvolver pro público "normal".

Há um tempo atrás eu acabei utilizando o Woocommerce como Rest Api e criei o frontend com NextJs.
O problema é que qualquer coisa além do woocommerce eu precisava criar um plugin novo.

Já pesquisei sobre alguns SAAS também, mas travei na mesma questão de customização...

Não encontraram nenhum projeto open source legal? Ou até mesmo criar um projeto open source? Pois acredito que muitos pensam na hipótese de criar a própria plataforma, mas acaba não compensando dependendo o tipo de cliente, aí criar algo open source seria mais fácil, pois reuniria pessoas com esse mesmo objetivo.

**Comment by karluslima** [comment]
> > > > > @matheusbronca Hoje ainda compensa desenvolver um? Soluções de terceiros não seriam mais interessantes? Estilo loja integrada ou tray.
> > > > 
> > > > 
> > > > Eu acredito que são opções diferentes, dá pra fugir de alguns custos e ter um pouco mais de liberdade na construção do e-commerce. Porém eu sou novato nesse negócio de e-commerce, então talvez eu esteja falando asneira, se houver pessoas com mais experiência que eu por favor colabore com uma resposta.
> > > 
> > > 
> > > Soluções de terceiros, geralmente são mais robustas e mais estáveis. Tudo depende do que o cliente precisa no e-commerce dele. Essa solução em next por exemplo, acredito que não atenderia as necessidades de players como Boticário, Renner, Decatlhon, Xiaomi, sem ter um desenvolvimento de longos meses e testes.
> > > Plataforma de e-commerce costumam ser fechadas quando se fala de back-end, então muitas vezes é mais complicado criar novas funcionalidades.
> > 
> > 
> > Grandes players ou varejistas se não optam por ter o próprio ecommerce acabam se associando a um marketplace (ou dependendo, ambos). E geralmente possuem fundos pra poder investir num próprio.
> > Creio que de um tempo pra cá pararam de fazer frameworks voltados a ecommerce, muitos realmente se apoiando em soluções SAAS. Faz tempo que não ouço falar nada de um "novo Magento", framework open que faz tudo (porém com core melhor programado 😆 )
> 
> Você sabe me informar quais as soluções SAAS mais populares aqui no Brasil?

Aqui no Brasil:

Linx Commerce
Vtex
NuvemShop
Tray
Vnda 
Jetcommerce 

Tem mais, todos variam preços e funcionalidades

**Comment by carlosviniciusananias** [comment]
> opa, falei errado auheuaheuhae mania de front, é SAAS mas acho que todo mundo entendeu rs
> 
> Englobem tb os outros, tipo Plataform As A Service quando falo sobre SAAS

As que atuei e conheço são 

Nuvemshop
Shopfy
Tray
Vtex _(vtex io)_
Loja integrada
Jet

Atualmente além de ter uma boa estrutura a vtex tem se atualizado muito. Hoje no vtex io praticamente se trabalha apenas com react, typescript e graphql. A performance do e-commerce é muito boa!

**Comment by ninetails** [comment]
Outro "front-only" que tropecei, em Svelte: [Litekart](https://github.com/itswadesh/sapper-ecommerce)
